### DISCUSSION TITLE
Health-check endpoints for monitoring and alerts

### DISCUSSION THREAD

**Discussion by vivekpulukuri** [specify/specify7#5863] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
A Health check endpoints to show the status of various specify components 
e.g., db, auth, cache, redis and etc so that can be integrated into enterprise alert systems such as nagios and etc

**Describe the solution you'd like**
django-health-check ?

**Reported By**
CSIRO

**Comment by maxpatiiuk** [comment]
This does not seem specific to specify - if you are running specify in docker, then any existing solution for health checking docker containers should work.

Quick Google search:
- https://hub.docker.com/r/pottava/docker-webui/
- https://docs.docksal.io/use-cases/portainer/
- https://www.reddit.com/r/docker/comments/zuyi12/looking_for_a_web_ui_management_for_docker/?rdt=61560

---

If you would rather use specify for this, you can use the `/context/system_info.json` endpoint (i.e https://sp7demofish.specifycloud.org/context/system_info.json) - if it returns 200, that tells you that web server, back-end and database are accessible. That is the endpoint we use for [Specify Cloud status page](https://updown.io/p/8g4mg)

If you would like to test auth status, you can get current user information from `https://sp7demofish.specifycloud.org/context/user.json` (returns 403 if not signed in. 200 if signed in or anonymous access is enabled)
